### PR TITLE
[AIRFLOW-4123] Add Exception handling for _change_state method in K8 …

### PR DESCRIPTION
**Pull request by andy-g14**
…Executor

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-4123) issues and references them in the PR title.
  "\[AIRFLOW-4123\] Add Exception handling for _change_state method in K8 Executor"
  - https://issues.apache.org/jira/browse/AIRFLOW-4123
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
  
   Failure to delete pod, causes multiple issue, like-

     1. The completed pod remains in the cluster forever,

     2. The taskInstance states are not updated in DB.

I want to retry _change_state in case of any failures. I am handling a generic exception, since APIException from kubernetes python client method delete_namespaced_pod does not encompass timeout exceptions like ReadTimeoutError or SocketError.

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/executors/kubernetes_executor.py
Comment (ramandumcs) [inline airflow/contrib/executors/kubernetes_executor.py]: Trailing "," is unnecessary

**Comment by feng-tao** [comment]
cc @dimberman

**Comment by dimberman** [comment]
@andy-g14 @feng-tao this LGTM but could you please tie this to a jira ticket?

**Comment by dimberman** [comment]
@feng-tao looking at the failing test, is this a race condition? Doesn't seem related to anything in the PR
 ```
test_no_orphan_process_will_be_left (tests.test_jobs.SchedulerJobTest) ... 
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

**Comment by feng-tao** [comment]
@dimberman , the test is pretty flaky. Are you able to merge pr now?

**Review by dimberman — APPROVED** [review]

**Comment by dimberman** [comment]
@feng-tao Seems my write access hasn't been turned on yet :(. Could I please have that set :)?

**Comment by feng-tao** [comment]
@dimberman , I think you need to use https://gitbox.apache.org/setup/ to setup your access which should take couples of hours to take effect.

**Comment by dimberman** [comment]
@feng-tao I now have the squash + merge ability. I'm gonna re-run the tests and if/when they pass I'll merge (first one, woo!).

**Comment by feng-tao** [comment]
@dimberman , I think as a committer, we could restart / trigger individual test(e.g only the failed test) instead of doing a full rebuild.

**Comment by ashb** [comment]
@dimberman Not sure if you have perms in Jira yet, but when merging a PR please also mark the issue as Resolved (not Closed) and set the "Fix Version" - either 2.0.0 if it's a big change/against new code in master, or 1.10.3 (currently) if you want the fix to be cherry-picked (by me usually) in to the release.

**Comment by dimberman** [comment]
@feng-tao Ahh good point! I've rerun the one failing test (this one seems to be around minikube not working correctly) and will merge if this one passes.

@ashb it doesn't seem that my jira perms have come in yet. Are there further steps I need to take or is that just a migration?

(thank you to you both for helping me get the hang of this btw :) )

**Comment by ashb** [comment]
@dimberman I've updated your Jira perms - try now.

**Comment by dimberman** [comment]
@ashb worked thank you!

I set it to 1.10.3 since it really shouldn't affect much else.

**Pull request merged**
